warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java

```diff
@@ -0,0 +1,51 @@
+package io.skai.warehouse.service.impl;
+
+import io.skai.warehouse.dto.ComponentDto;
+import io.skai.warehouse.service.ComponentService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+@Testcontainers
+@SpringBootTest
+class ComponentServiceImplTest {
+
+    private static final String SCREEN = "screen";
+    private static final String KEYBOARD = "keyboard";
+    private static final String TOUCHPAD = "touchpad";
+    @Container
+    private static final MySQLContainer container = new MySQLContainer<>("mysql:latest");
+    @Autowired
+    private ComponentService componentService;
+
+    @DynamicPropertySource
+    public static void overrideProperties(DynamicPropertyRegistry registry) {
+        registry.add("spring.datasource.url", container::getJdbcUrl);
+        registry.add("spring.datasource.username", container::getUsername);
+        registry.add("spring.datasource.password", container::getPassword);
+    }
+
+    @Test
+    void shouldWriteToDbThenReturnIt() {
+        List<ComponentDto> inputData = getInputData();
+        List<ComponentDto> result = componentService.create(inputData);
+        assertThat(result).hasSize(3)
+                .allMatch(n -> n.id() > 0)
+                .extracting("name")
+                .containsExactlyInAnyOrder(TOUCHPAD, KEYBOARD, SCREEN);
```

Comment:
if first check fail, this wouldn't be verified, every validation in test should be atomic.

Reply:
if i write it separately like this
`assertThat(result).hasSize(3)
                .allMatch(componentDto -> componentDto.id() > 0);
        assertThat(result)
                .extracting("name")
                .containsExactlyInAnyOrder(TOUCHPAD, KEYBOARD, SCREEN);` 
part with "extracting" will be not checked too if id checks fill fail. I wan't to check condition : ID were generated(>0) AND name are same as id input parameters so i used one check.